src/patient-registration/patient-registration.component.tsx

```diff
@@ -77,16 +100,45 @@ export const PatientRegistration: React.FC = () => {
     return () => abortController.abort();
   }, []);
 
+  useEffect(() => {
+    const abortController = new AbortController();
+    getPrimaryIdentifierType(abortController).then(primaryIdentifierType => {
+      getSecondaryIdentifierTypes(abortController).then(secondaryIdentifierTypes => {
+        let idTypes = [];
+        idTypes = [primaryIdentifierType, ...secondaryIdentifierTypes].filter(Boolean);
+        idTypes.forEach(type => {
+          // update form initial values and validation schema
+          initialFormValues[type.fieldName] = '';
+          let fieldValidationProps = Yup.string();
+          if (type.required) {
+            fieldValidationProps = fieldValidationProps.required("Identifier can't be blank!");
+          }
+          if (type.format) {
+            fieldValidationProps = fieldValidationProps.matches(new RegExp(type.format), 'Invalid identifier format!');
+          }
+          validationSchema[type.fieldName] = fieldValidationProps;
+        });
+        setIdentifierTypes(idTypes);
+      }, createErrorHandler());
+    }, createErrorHandler());
+    return () => abortController.abort();
+  }, []);
+
   const onFormSubmit = (values: FormValues) => {
+    const patIdentifiers = identifierTypes.reduce(function(ids, id) {
```

Comment:
`pt` is short for "patient" (though outside medicine is also often "point" ). `pat` in the OpenMRS context is "Patient Attribute Type", so let's avoid that. But here you can just call this variable `identifiers`; there's no ambiguity as to what they correspond to.

Reply:
Resolved